### PR TITLE
ALTAPPS-770: Shared beta projects are not shown in project list

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[#ALTAPPS-770](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-770)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Add fetching beta projects for beta users in `ProjectSelectionListActionDispatcher`;
- Implement `sortByScore` extension to sort projects;
- Add `sortedProjectsIds` to the state;
- Add `ProjectListItem.isBeta`;
- Add sorting tests;
- Support `beta` badge on both platforms.

### CODE REVIEW

File: shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListFeature.kt [hunk 1]
```diff
@@ -79,6 +82,7 @@ object ProjectSelectionListFeature {
 
     internal sealed interface ContentFetchResult : Message {
         data class Success(
+            val profile: Profile,
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListFeature.kt:85]: I think we can delete this, I don't see any further usages

**Review by ivan-magda — COMMENTED** [review]
Looks nice, just a minor comments

File: shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListFeature.kt (hunk 1 above)
Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListFeature.kt:85]: It is used in the reducer. See [ProjectSelectionListReducer](https://github.com/hyperskill/mobile-app/blob/08720628660312b42e40350e78c710b4ff4824cc/shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListReducer.kt#L149-L153)

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProjectSelectionList/Views/Grid/Cell/ProjectSelectionListGridCellView.swift [hunk 2]
```diff
@@ -81,7 +81,8 @@ struct ProjectSelectionListGridCellView_Previews: PreviewProvider {
             isBestRated: true,
             isIdeRequired: true,
             isFastestToComplete: true,
-            isCompleted: true
+            isCompleted: true,
+            isBeta: true
```
[the remark this answers is not in the document]
Comment (ivan-magda) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProjectSelectionList/Views/Grid/Cell/ProjectSelectionListGridCellView.swift:85]: > TODO: support on iOS

This is a comment for me, I'll render beta badge on iOS

File: shared/src/commonMain/kotlin/org/hyperskill/app/profile/domain/model/Profile.kt [hunk 3]
```diff
@@ -42,8 +42,14 @@ data class Profile(
     @SerialName("track_id")
     val trackId: Long?,
     @SerialName("track_title")
-    val trackTitle: String?
-)
+    val trackTitle: String?,
+    @SerialName("is_beta")
+    val isBeta: Boolean,
+    @SerialName("features")
+    val features: Map<String, Boolean>
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/profile/domain/model/Profile.kt]: @ivan-magda 
I've made `profile.features` field private and provided access to its wrapper - `FeaturesMap`. `FeaturesMap` is used as a more concrete type to be an extensions receiver.

File: shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListActionDispatcher.kt [hunk 4]
```diff
@@ -111,4 +58,86 @@ class ProjectSelectionListActionDispatcher(
...
+
+            val profile = profileInteractor.getCurrentProfile(DataSourceType.CACHE)
+                .getOrElse {
+                    sentryInteractor.finishTransaction(transaction, throwable = it)
+                    onNewMessage(ProjectSelectionListFeature.ContentFetchResult.Error)
+                    return@coroutineScope
+                }
+
+            val track =
+                trackRepository.getTrack(action.trackId, action.forceLoadFromNetwork)
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListActionDispatcher.kt:79]: @ivan-magda 
In my opinion, we should create & use interaction for features or common use cases.
In this case, I suggest creating a dedicated `ProjectSelectionListInteractor` which under the hood will refer to repositories.

[the remark this answers is not in the document]
Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListActionDispatcher.kt:80]: @ivan-magda TODO: create a `dept` task

**Review by ivan-magda — APPROVED** [review]

File: shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListActionDispatcher.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/project_selection/presentation/ProjectSelectionListActionDispatcher.kt:80]: https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-787

**Pull request merged**